splink/comparison_template_library.py

```diff
@@ -0,0 +1,173 @@
+# Script defining comparison templates that act as wrapper functions which produce
+# comparisons based on the type of data in a column with default values to make
+# it simpler to use splink out-of-the-box
+
+from __future__ import annotations
+
+from .comparison import Comparison  # change to self
+from .comparison_library_utils import datediff_error_logger
+from .misc import ensure_is_iterable
+
+
+class DateComparisonBase(Comparison):
+    def __init__(
+        self,
+        col_name,
+        include_exact_match_level=True,
+        term_frequency_adjustments=False,
+        separate_1st_january=True,
+        levenshtein_thresholds=[2],
+        datediff_thresholds=[[1, 1], ["month", "year"]],
+        m_probability_exact_match=None,
+        m_probability_1st_january=None,
+        m_probability_or_probabilities_lev: float | list = None,
+        m_probability_or_probabilities_datediff: float | list = None,
+        m_probability_else=None,
+        include_colname_in_charts_label=False,
+    ):
+        """A wrapper to generate a comparison for a date column the data in
+        `col_name` with preselected defaults.
+
+        The default arguments will give a comparison with comparison levels:\n
+        - Exact match (1st of January only)\n
+        - Exact match (all other dates)\n
+        - Levenshtein distance <= 2\n
+        - Date difference <= 1 month\n
+        - Date difference <= 1 year \n
+        - Anything else
+
+        Args:
+            col_name (str): The name of the column to compare
+            include_exact_match_level (bool, optional): If True, include an exact match
+                level. Defaults to True.
+            term_frequency_adjustments (bool, optional): If True, apply term frequency
+                adjustments to the exact match level. Defaults to False.
+            separate_1st_january (bool, optional): If True, include a separate
+                comparison level for 1st January.
```

Comment:
Perhaps add a little more information here. Is it an exact match + check on 1st Jan?